xdsl/dialects/mesh.py

```diff
@@ -1,18 +1,75 @@
+from collections.abc import Sequence
+from typing import ClassVar, TypeAlias
+
 from xdsl.dialects.builtin import (
+    I16,
     I64,
+    ArrayAttr,
+    BytesAttr,
     DenseArrayBase,
     SymbolNameConstraint,
+    i16,
 )
 from xdsl.dialects.utils.dimension_list import DimensionList
-from xdsl.ir import Dialect, VerifyException
+from xdsl.ir import (
+    Attribute,
+    Dialect,
+    OpaqueSyntaxAttribute,
+    ParametrizedAttribute,
+    VerifyException,
+)
 from xdsl.irdl import (
     IRDLOperation,
+    irdl_attr_definition,
     irdl_op_definition,
     prop_def,
     traits_def,
 )
+from xdsl.parser import AttrParser
+from xdsl.printer import Printer
 from xdsl.traits import SymbolOpInterface
 
+MeshAxesAttr: TypeAlias = DenseArrayBase[I16]
+
+
+@irdl_attr_definition
+class MeshAxesArrayAttr(ParametrizedAttribute, OpaqueSyntaxAttribute):
+    name = "mesh.axisarray"
+
+    axes: ArrayAttr[MeshAxesAttr]
+
+    ELT_TYPE: ClassVar = i16
+
+    @classmethod
+    def parse_parameters(cls, parser: AttrParser) -> Sequence[Attribute]:
+        def parse_mesh_axes_attr():
+            elements = parser.parse_comma_separated_list(
+                parser.Delimiter.SQUARE,
+                parser.parse_integer,
+            )
+
+            return DenseArrayBase[I16](
+                cls.ELT_TYPE, BytesAttr(cls.ELT_TYPE.pack(elements))
+            )
```

Comment:
Can you please move this out to the top level, probably as a private function (with a short doc string describing what it does)